From d17f5f9afc208649ba918b25207370ad0442105b Mon Sep 17 00:00:00 2001
From: Kevin McCarthy <kevin@8t8.us>
Date: Fri, 29 Apr 2016 18:00:10 -0700
Subject: [PATCH 22/37] Remove newline insertion from pager. (closes #3835)

The window patches commited now manually move to the next line, so
there is no need to insert newlines for that purpose.

The newlines also goof windows placed to the right of the pager, as is
the case for the (currently external) sidebar-on-the-right
functionality.

Thanks to Richard Russon for the original patch and help testing the
revised patch.
---
 pager.c | 13 +++----------
 1 file changed, 3 insertions(+), 10 deletions(-)

diff --git a/pager.c b/pager.c
index 96cc506..1aebaec 100644
--- a/pager.c
+++ b/pager.c
@@ -1510,18 +1510,12 @@ display_line (FILE *f, LOFF_T *last_pos, struct line_t **lineInfo, int n,
     ATTRSET(def_color);
   }
 
-  /* ncurses always wraps lines when you get to the right side of the
-   * screen, but S-Lang seems to only wrap if the next character is *not*
-   * a newline (grr!).
-   */
-#ifndef USE_SLANG_CURSES
-    if (col < pager_window->cols)
-#endif
-      addch ('\n');
+  if (col < pager_window->cols)
+    mutt_window_clrtoeol (pager_window);
 
   /*
    * reset the color back to normal.  This *must* come after the
-   * addch('\n'), otherwise the color for this line will not be
+   * clrtoeol, otherwise the color for this line will not be
    * filled to the right margin.
    */
   if (flags & M_SHOWCOLOR)
@@ -1851,7 +1845,6 @@ mutt_pager (const char *banner, const char *fname, int flags, pager_t *extra)
 	mutt_window_clrtoeol (pager_window);
 	if (option (OPTTILDE))
 	  addch ('~');
-	addch ('\n');
 	lines++;
         mutt_window_move (pager_window, lines, 0);
       }
-- 
2.9.0

